MapleServer2/Types/Inventory.cs

```diff
@@ -45,10 +45,10 @@ public class Inventory {
             // Item has a slot set, try to use that slot
             if (item.Slot >= 0) {
                 if (!SlotTaken(item, item.Slot)) {
-                    //Performs a item exist check before adding item
-                    if (Check(item))
+                    // Performs a item exist check before adding item
+                    if (Exists(item))
                     {
-                        return false;
+                        return Exists(item);
```

Comment:
This is just repetitive and waste of processing, revert to return false or add a variable above the if with the value of `Exists(item)`

Reply:
I would extract it into a method addItem and reuse since the code fragments are basically the same